share/completion/git-restore

```diff
@@ -0,0 +1,48 @@
+# (C) 2024 unrealapex
+
+# Completion script for the "git-restore" command.
+
+function completion/git-restore {
+        WORDS=(git restore "${WORDS[2,-1]}")
+        command -f completion//reexecute
+}
+
+function completion/git::restore:arg {
+
+        OPTIONS=( #>#
+        "s --source=; restore the working tree files with the content from the given tree"
+        "p --patch; interactively select hunks in the difference between the restore source and the restore location"
+        "W --worktree; specify the restore location"
+        "S --staged; specify the restore location"
+        "q --quiet; quiet, supress feedback messages"
+        "--progress; display progress status"
+        "--no-progress; do not display progress status"
+        "--ours; use stage #2 (ours) when restoring files in the working tree from the index"
+        "--theirs; use stage #3 (theirs) when restoring files in the working tree from the index"
+        "m --merge; when restoring files on the working tree from the index, recreate the conflicted merge in the unmerged paths"
+        "--conflict=; the same as the --merge option, but changes the way the conflicting hunks are presented, overriding the merge.conflictStyle configuration variable"
+        "--ignore-unmerged; when restoring files on the working tree from the index, do not abort the operation if there are unmerted entries and neither --ours, --theirs, --merge, or --conflict is specified"
+        "--ignore-skip-worktree-bits; in sparse checkout mode, the default is to only update entries matched by <pathspec> and sparse patterns in \$GIT_DIR/info/sparse-checkout."
+        "--recurse-submodules; if <pathspec> names an active submodule and the restore location includes the working tree"
+        "--no-recurse-submodules"
+        "--overlay; never removefiles when restoring."
+        "--no-overlay; remove tracked files that do not appear in the --source tree"
+        "--pathspec-from-file=; pathspec is passed in <file> instead of commandline args"
+        "--pathspec-file-nul; only meaningful with --pathspec-from-file"
+        "--; do not interpret any more arguments as options"
+        ) #<#
+
+        command -f completion//parseoptions -n
+        case $ARGOPT in
+                (-)
+                        command -f completion//completeoptions
+                        ;;
+                ('')
+                        command -f completion/git::completerefpath
```

Comment:
`git restore` only accepts pathnames, not refs. We have a separate function for this case.

```suggestion
                        command -f completion/git::completepath
```